### PR TITLE
experimental web apigw cleanup

### PR DESCRIPTION
**Pull request by drcariel**
I was planning on consolidating things like the following into a single var accross the entire project:
```
apigw_host: "http://{{ groups['apigateway']|first }}:{{apigateway.port.api}}/v1"
apigw_host_v2: "http://{{ groups['apigateway']|first }}:{{apigateway.port.api}}/v2"
```

However, the implications seems to be far reaching, and I am not sure the most efficient way to go about it.

Should I be retaining the v1 in all the routes? should it be all updated to v2? or the version removed entirely?

@csantanapr @mdeuser

### CODE REVIEW

**Comment by csantanapr** [comment]
David what you have is not directly related to apigateway.
This PR has two concepts that should be split.
1. apigateway experimental v1
2. `/experimental/web`  was the first implementation on what we talk today as web actions.

I think @rabbah would be happy to see `/experimental/web` removed. 🌮 
I would delegate to him on to review and assist on this PR. 🙏 

I would suggest since the majority of this PR is related to `experimental/web` create a new PR and move the changes to the apigateway.md to a new PR, that's the only  related change to apigateway in this PR.

Then on the new PR for the ansible:
1. remove `apigw_host: "http://{{ groups['apigateway']|first }}:{{apigateway.port.api}}/v1"`
all the environments group vars
2. remove from whisk.properties ansible template https://github.com/apache/incubator-openwhisk/blob/master/ansible/templates/whisk.properties.j2#L103

**Comment by rabbah** [comment]
can def nuke /experimental and all the stuff that goes with it. go for it.

**Comment by drcariel** [comment]
sister PR in Blue ~3909

File: tests/src/test/scala/whisk/core/cli/test/WskWebActionsTests.scala [hunk 1]
```diff
@@ -47,11 +47,6 @@ import whisk.core.entity.Subject
  * Tests web actions.
  */
 @RunWith(classOf[JUnitRunner])
-class WskWebActionsTestsV1 extends WskWebActionsTests {
-    override val testRoutePath = "/api/v1/experimental/web"
-}
-
-@RunWith(classOf[JUnitRunner])
 class WskWebActionsTestsV2 extends WskWebActionsTests with BeforeAndAfterAll {
     override val testRoutePath = "/api/v1/web"
```
Comment (rabbah) [inline tests/src/test/scala/whisk/core/cli/test/WskWebActionsTests.scala]: +1 to rename class and remove v2.

**Comment by mdeuser** [comment]
@drcariel - the `apigw_host: "http://{{ groups['apigateway']|first }}:{{apigateway.port.api}}/v1"` path won't be used any more with the removal of the v1 interface.  there is likely v1-only code in the routemgmt actions (getApi, deleteApi, createApi)

File: core/controller/src/main/scala/whisk/core/controller/WebActions.scala [hunk 2]
```diff
@@ -74,14 +74,6 @@ protected[controller] sealed class WebApiDirectives private (prefix: String) {
 protected[controller] object WebApiDirectives {
```
Comment (rabbah) [inline core/controller/src/main/scala/whisk/core/controller/WebActions.scala]: i suggest you remove this entire class - and change the constructor for WebApiDirectives:

```
protected[controller] sealed class WebApiDirectives private (prefix: String = "__ow_") {
```

File: tests/src/test/scala/whisk/core/cli/test/WskWebActionsTests.scala (hunk 1 above)
Comment (drcariel) [inline tests/src/test/scala/whisk/core/cli/test/WskWebActionsTests.scala]: should I do the same with WebActionsApiTests? I don't think it's as easy

**Comment by dubeejw** [comment]
Are we removing the experimental API gateway or that _and_ experimental web actions?

File: core/controller/src/main/scala/whisk/core/controller/RestAPIs.scala [hunk 3]
```diff
@@ -197,8 +197,7 @@ class RestAPIVersion(config: WhiskConfig, apiPath: String, apiVersion: String)(
     private val triggers = new TriggersApi(apiPath, apiVersion)
     private val activations = new ActivationsApi(apiPath, apiVersion)
     private val rules = new RulesApi(apiPath, apiVersion)
-    private val webexp = new WebActionsApi(Seq("experimental", "web"), WebApiDirectives.exp)
-    private val web = new WebActionsApi(Seq("web"), WebApiDirectives.web)
+    private val web = new WebActionsApi(Seq("web"), WebApiDirectives)
```
Comment (drcariel) [inline core/controller/src/main/scala/whisk/core/controller/RestAPIs.scala]: @rabbah I am having trouble accessing WebApiDirectives here now that I made your suggested edit to the constructor, any suggestions?

Comment (rabbah) [inline core/controller/src/main/scala/whisk/core/controller/RestAPIs.scala]: remove `sealed` on the class.

Comment (drcariel) [inline core/controller/src/main/scala/whisk/core/controller/RestAPIs.scala]: ok, I assumed the class needed to remain sealed, thank you

**Comment by drcariel** [comment]
PG1~1936

File: tests/src/test/scala/whisk/core/cli/test/WskWebActionsTests.scala [hunk 4]
```diff
@@ -47,94 +47,14 @@ import whisk.core.entity.Subject
  * Tests web actions.
  */
 @RunWith(classOf[JUnitRunner])
-class WskWebActionsTestsV1 extends WskWebActionsTests {
-    override val testRoutePath = "/api/v1/experimental/web"
-}
-
-@RunWith(classOf[JUnitRunner])
-class WskWebActionsTestsV2 extends WskWebActionsTests with BeforeAndAfterAll {
```
Comment (rabbah) [inline tests/src/test/scala/whisk/core/cli/test/WskWebActionsTests.scala]: these tests should stay. is this just moved down verbatim?

Comment (drcariel) [inline tests/src/test/scala/whisk/core/cli/test/WskWebActionsTests.scala]: Yes, the tests are all still there. Just renamed and consolidated

Comment (drcariel) [inline tests/src/test/scala/whisk/core/cli/test/WskWebActionsTests.scala]: See lines 251 - 317

**Comment by rabbah** [comment]
@drcariel applied `wip` label for you - what's left?

**Comment by drcariel** [comment]
@rabbah not sure, @csantanapr asked me to change the title since he wanted me to focus on another PR.

**Comment by dubeejw** [comment]
@rabbah, looks like people are still using experimental web actions. Should at least send out a notifications above its removal in advance.

**Pull request merged**